src/test/java/teammates/test/cases/datatransfer/AccountAttributesTest.java

```diff
@@ -71,9 +72,12 @@ public void testGetEntityTypeAsString() {
     @Test
     public void testToEntity() {
         AccountAttributes account = createValidAccountAttributesObject();
-        Account expectedAccount = new Account(account.googleId, account.name, account.isInstructor,
-                account.email, account.institute, StudentProfileAttributes.builder().build().toEntity());
+        Account expectedAccount = new Account(account.googleId, account.name,
+                account.isInstructor, account.email, account.institute,
+                StudentProfileAttributes.builder(account.googleId).build().toEntity());
 
+        // save studentProfile to datastore so that it can be retrieved later
+        new ProfilesDb().saveEntity(account.studentProfile.toEntity());
```

Comment:
I realised this isn't actually necessary. There's a much simpler way to construct `actualAccount` that doesn't try to retrieve the student profile entity from the database.

Reply:
Hm, I'm not sure that the problem lies only with the `actualAccount` constructor (although using `Account actualAccount = account.toEntity()` is probably better; will make the necessary changes!). Checking for equality between `expectedAccount` and `actualAccount` requires calling `getStudentProfile()` which in turn uses `get()` which looks in the datastore since we're no longer storing `studentProfile`s locally in the `Account`.

Reply:
Well `account.toEntity()` passes. Might be because the student profile is null?

Reply:
`account.toEntity()` works because it constructs the Account entity from an `AccountAttributes` object which stores a `StudentProfileAttributes` object so no datastore references are needed. But `actualAccount.toEntity()` creates an `Account` object that stores a `Ref<StudentProfile>` instead of saving the actual `StudentProfile` or `StudentProfileAttributes`, so when the test tries to compare the StudentProfileAttributes later there's a null pointer error (around line 85).